src/requirements/requirement-graph-builder.ts

```diff
@@ -12,86 +18,97 @@ type BuildRequirementFulfillmentGraphParameters<Requirement extends string, Cour
   readonly userCourses: readonly Course[];
   /**
    * Some requirements might have several different ways to fulfill them. This is a map from
-   * toggleable requirements to a list of courses that could be applied to the requirement under
+   * toggleable requirements to a list of course IDs that could be applied to the requirement under
    * the user's choice.
    */
-  readonly userChoiceOnFulfillmentStrategy: Readonly<Record<Requirement, readonly Course[]>>;
+  readonly userChoiceOnFulfillmentStrategy: Readonly<Record<Requirement, readonly number[]>>;
   /**
-   * A list of (requirement, course) tuple that describe how the user decides how a course is used to
-   * satisfy requirements. Suppose a course c can be used to satisfy both r1 and r2, but the
-   * requirements do not allow double counting, then a tuple (r1, c) means that we should keep the
+   * The mapping from course's unique ID to requirement.
+   * It describes how the user decides how a course is used to satisfy requirements.
+   * Suppose a course with unique ID c can be used to satisfy both r1 and r2, but the
+   * requirements do not allow double counting, then a mapping `c => r1` means that we should keep the
    * (r1, c) edge in the graph and drop the (r2, c) edge.
    */
-  readonly userChoiceOnDoubleCountingElimination: readonly (readonly [Requirement, Course])[];
-  /** See RequirementFulfillmentGraph for its usage. */
-  readonly getCourseUniqueID: (c: Course) => string | number;
+  readonly userChoiceOnDoubleCountingElimination: Readonly<Record<number, Requirement>>;
   /**
    * Naively give a list of courses that can satisfy a requirement. Most of the time this function
    * should just return the pre-computed course list. For requirements have multiple fulfillment
    * strategies, it will return the union of all pre-computed course list.
    */
   readonly getAllCoursesThatCanPotentiallySatisfyRequirement: (
     requirement: Requirement
-  ) => readonly Course[];
+  ) => readonly number[];
```

Comment:
can we update the spec so that it says we're getting a list of the course id (crseId from cornell course roster api that crosslisted courses share)